engine/src/main/java/org/destinationsol/game/SolGame.java

```diff
@@ -113,15 +114,15 @@
     private HullConfig respawnHull;
     private boolean isPlayerRespawned;
 
-    public SolGame(SolApplication cmp, String shipName, boolean tut, boolean isNewGame, CommonDrawer commonDrawer) {
-        solApplication = cmp;
+    public SolGame(SolApplication cmp, String shipName, boolean tut, boolean isNewGame, CommonDrawer commonDrawer, Context context) {
+        this.context = context;
```

Comment:
A context should not be cached in a field; instead, a context must be passed to the constructor and the instances required from the context must be retrieved and cached in fields. In this case, the `SolApplication` instance (used later in the class) in the context must be retrieved (`context.get(SolApplication.class)`) and cached in a field.